### PR TITLE
Safety as an Action

### PR DESCRIPTION
**Pull request by cb22**
Bring in changes to allow running Safety easily as a GitHub Action.

### CODE REVIEW

File: entrypoint.sh [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+    elif [[ $(basename ${SAFETY_ACTION_REQUIREMENTS}) == "poetry.lock" ]]; then
+        SAFETY_OS_DESCRIPTION="${SAFETY_OS_DESCRIPTION} file_poetry_converted"
+
+        cd "$(dirname ${SAFETY_ACTION_REQUIREMENTS})"
+        poetry export -f requirements.txt --without-hashes > /tmp/requirements.txt
+        echo "[Safety Action] Converted ${SAFETY_ACTION_REQUIREMENTS} to /tmp/requirements.txt using poetry." 1>&2
+        SAFETY_ACTION_REQUIREMENTS="/tmp/requirements.txt"
+    fi
+
+    if [[ "${SAFETY_ACTION_CONTINUE_ON_ERROR}" != "" ]]; then
```
Comment (yeisonvargasf) [inline entrypoint.sh:142]: @cb22, a whitespace may change the behavior in the action, I think we could improve the check here.

**Review by yeisonvargasf — APPROVED** [review]
Great work! LGTM, I left a few comments.

File: .github/workflows/test-insecure.yml [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+
+      - run: python -m pip install -r tests/action/requirements.txt-insecure
+
+      - uses: ./
+        id: scan-4
+        continue-on-error: true
+        with:
+          api-key: ${{ secrets.SAFETY_API_KEY }}
+
+      - if: steps.scan-4.outcome != 'failure' || steps.scan-4.outputs.exit-code != '64'
```
[the remark this answers is not in the document]
Comment (cb22) [reply .github/workflows/test-insecure.yml:91]: @yeisonvargasf ah right so this `continue-on-error` is passed to the GitHub action, not to Safety. So in this case, it's just telling the action to continue, and we fail later if the previous step didn't fail

File: Dockerfile [hunk 3]
```diff
@@ -1,3 +1,21 @@
...
-CMD ["python"]
+FROM python:3.10-slim
+
+# Don't use WORKDIR here as per Github's docs
+RUN mkdir /app
+
+RUN apt-get update && apt-get -y install docker.io jq && apt-get clean && rm -rf /var/lib/apt/lists/*
+
+# Install poetry and pipenv; used for converting their respective lockfile formats to generic requirements.txt
+RUN cd /app && python3 -m pip install poetry==1.1.13 pipenv==2022.5.2
```
[the remark this answers is not in the document]
Comment (cb22) [reply Dockerfile:9]: Good point, will update!

File: entrypoint.sh (hunk 1 above)
Comment (cb22) [reply entrypoint.sh:142]: @yeisonvargasf yes true. We could make it accept (yes|true), case insensitive?

Comment (yeisonvargasf) [reply entrypoint.sh:142]: Agree!

**Pull request merged**